src/WebJobs.Script.Abstractions/Description/Retry.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Script.Abstractions.Description
+{
+    public class Retry
+    {
+        public RetryStrategy Strategy { get; set; }
```

Comment:
Wondering if we should have this strategy property. We could instead infer based on properties user has set, so they don't have to specify this? What do you think @brettsam ?

Reply:
Then the rules would be:
1. If you explicitly set DelayInterval, it's Fixed
2. If you explicitly set Minimum or Maximum, then it's Exponential.

What if you don't set any? Or what if there's a conflict in the rules?

Reply:
Yes, I was thinking something like that. Note that we still have similar validation work to do even we had them specify a strategy value, e.g. they might declare "fixedDelay" but not specify any other values. @pragnagopa how are you handling validation - where will it fail in cases like this?

Anyhow, maybe it's OK to just be explicit here and have a strategy property. Can we name it "retry" rather than "functionRetry" in function.json? If we ever supported functionTimeout specification in function.json, we'd also probably want to call that just "timeout", rather than "functionTimeout" as in host.json.

```javascript
"retry": {
  "strategy": "fixedDelay",
  "maxRetryCount": 3,
  "delayInterval": "00:00:10"
}

"retry": {
  "strategy": "exponentialBackoff",
  "maxRetryCount": 3,
  "minimumInterval": "00:00:02",
  "maximumInterval": "00:00:30"
}
```

Reply:
I prefer having `strategy` specified explicitly - bit more clear on which values to specify.  Validation for these values comes from underlying strategies - https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Timers/RandomizedExponentialBackoffStrategy.cs and https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Timers/FixedDelayStrategy.cs

Reply:
If `interval` is not specified, it defaults to `00:00:00` - is this ok? or should we enforce specifying non-zero delay for each strategy?

Reply:
Let's require them to specify it explicitly. the underlying SDK FixedRetryAttribute requires a value to be explicitly provided on construction right? This would be consistent with that.

Reply:
Fixed!